taipy/core/common/_check_dependencies.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021-2024 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+from importlib import util
+
+
+def _check_dependency_is_installed(module_name: str, package_name: str) -> None:
+    """
+        Check if a package is installed.
+
+        Args:
+            module_name: Name of the taipy module importing the package.
+            package_name: Name of the package.
+    .
+    """
+    extras = {
+        "boto3": "s3",
+        "pymongo": "mongo",
+    }
+    if not util.find_spec(package_name):
+        raise RuntimeError(
+            f"Cannot use {module_name} as {package_name} package is not installed. Please install it  "
+            f"using `pip install taipy[{extras.get(package_name)}]`."
```

Comment:
Should we check `if extras.get(package_name):` first?

`pip install taipy[None]` is not ideal when we forgot to add the package name to the `extras`

Reply:
Oh really?? Don't bother with such a strange idea!

Reply:
That shouldn't happen, we are the only ones using this kind of function we control which values are passed.